pkg/gateway/backup/handler/grpc/handler.go

```diff
@@ -0,0 +1,705 @@
+//
+// Copyright (C) 2019-2020 Vdaas.org Vald team ( kpango, rinx, kmrmt )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package grpc provides grpc server logic
+package grpc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/vdaas/vald/apis/grpc/v1/payload"
+	"github.com/vdaas/vald/apis/grpc/v1/vald"
+	client "github.com/vdaas/vald/internal/client/v1/client/gateway/vald"
+	"github.com/vdaas/vald/internal/errgroup"
+	"github.com/vdaas/vald/internal/errors"
+	"github.com/vdaas/vald/internal/info"
+	"github.com/vdaas/vald/internal/log"
+	"github.com/vdaas/vald/internal/net/grpc"
+	"github.com/vdaas/vald/internal/net/grpc/status"
+	"github.com/vdaas/vald/internal/observability/trace"
+	"github.com/vdaas/vald/internal/safety"
+	"github.com/vdaas/vald/pkg/gateway/backup/service"
+	"github.com/vdaas/vald/pkg/gateway/internal/location"
+)
+
+type server struct {
+	eg                errgroup.Group
+	backup            service.Backup
+	gateway           client.Client
+	copts             []grpc.CallOption
+	streamConcurrency int
+}
+
+const apiName = "vald/gateway-backup"
+
+func New(opts ...Option) vald.Server {
+	s := new(server)
+
+	for _, opt := range append(defaultOpts, opts...) {
+		opt(s)
+	}
+	return s
+}
+
+func (s *server) Exists(ctx context.Context, meta *payload.Object_ID) (*payload.Object_ID, error) {
+	ctx, span := trace.StartSpan(ctx, apiName+".Exists")
+	defer func() {
+		if span != nil {
+			span.End()
+		}
+	}()
+	ips, err := s.backup.GetLocation(ctx, meta.GetId())
+	if err != nil {
+		log.Debug("an error occurred during calling meta Exists:", err)
+		return s.gateway.Exists(ctx, meta, s.copts...)
+	}
+	if len(ips) > 0 {
+		return meta, nil
+	}
+	return nil, status.WrapWithNotFound(fmt.Sprintf("Exists API meta %s's uuid not found", meta.GetId()), err, meta.GetId(), info.Get())
+}
+
+func (s *server) Search(ctx context.Context, req *payload.Search_Request) (res *payload.Search_Response, err error) {
+	ctx, span := trace.StartSpan(ctx, apiName+".Search")
+	defer func() {
+		if span != nil {
+			span.End()
+		}
+	}()
+	if len(req.Vector) < 2 {
+		return nil, errors.ErrInvalidDimensionSize(len(req.Vector), 0)
```

Comment:
is it better to write like this?
```suggestion
	if len(req.GetVector()) < 2 {
		return nil, errors.ErrInvalidDimensionSize(len(req.GetVector()), 0)
```